nevergrad/functions/images/core.py

```diff
@@ -47,17 +50,12 @@ def __init__(self, problem_name: str = "recovering", index: int = 0) -> None:
         array.set_bounds(lower=0, upper=255.99, method="clipping", full_range_sampling=True)
         max_size = ng.p.Scalar(lower=1, upper=200).set_integer_casting()
         array.set_recombination(ng.p.mutation.Crossover(axis=(0, 1), max_size=max_size)).set_name("")  # type: ignore
-        super().__init__(self._loss, array)
+        super().__init__(loss(reference=self.data), array)
+        self.add_descriptors(loss=loss.__class__.__name__)
+        self.loss_function = loss(reference=self.data)
 
     def _loss(self, x: np.ndarray) -> float:
-        assert self.problem_name == "recovering"
-        x = np.array(x, copy=False).ravel()
-        x = x.reshape(self.domain_shape)
-        assert x.shape == self.domain_shape, f"Shape = {x.shape} vs {self.domain_shape}"
-        # Define the loss, in case of recovering: the goal is to find the target image.
-        assert self.index == 0
-        value = float(np.sum(np.fabs(np.subtract(x, self.data))))
-        return value
+        return self.loss_function(x)
```

Comment:
please delete the `_loss` function which is not used anymore

Reply:
sure. Sorry I thought it was already done.